### PR TITLE
Billboard sprite shader - fix clipping in 3d models

### PR DESCRIPTION
**Pull request by Jlabarca**
image

Billboard logic moved into a shader which also avoids clipping into 3D models due it rotation
Based in "bgolus"  solution at https://forum.unity.com/threads/problem-solving-2d-billboard-sprites-clipping-into-3d-environment.680374/#post-6863138

WIP - characters head anchor positions looking wrong in some directions

### CODE REVIEW

File: UnityClient/Assets/Scripts/Renderer/GridRenderer.cs [hunk 1]
```diff
@@ -50,7 +50,7 @@ public class GridRenderer : MonoBehaviour {
     }
 
     private void LoadGridTexture() {
-        material = new Material(Shader.Find("Unlit/WalkableShader"));
+        material = new Material(Shader.Find("Universal Render Pipeline/2D/Sprite-Unlit-Default"));
```
Comment (Jlabarca) [inline UnityClient/Assets/Scripts/Renderer/GridRenderer.cs:53]: This is just to have a valid shader in the grid - probably temporary until we update WalkableShader

Comment (Danil0v3s) [reply UnityClient/Assets/Scripts/Renderer/GridRenderer.cs:53]: Don't worry about that, there's nothing fancy about the walkable shader, I don't even remember what it does btw 😅

File: UnityClient/Assets/Resources/Shaders/Sprites/BillboardSpriteShader.shader [hunk 2]
```diff
@@ -0,0 +1,101 @@
+Shader "UnityRO/BillboardSpriteShader"
```
Comment (Danil0v3s) [inline UnityClient/Assets/Resources/Shaders/Sprites/BillboardSpriteShader.shader:1]: Why isn't this billboard shader not using the commonly used Billboard.cginc?

Comment (Jlabarca) [reply UnityClient/Assets/Resources/Shaders/Sprites/BillboardSpriteShader.shader:1]: Did not know about a commonly used Billboard.cginc, We could check what type of rotation towards camera it does, probably something similar

**Comment by Danil0v3s** [comment]
Had a chance to test this and it looks good. In all my ignorance the problem you mentioned seems to reside within `rayPlaneIntersection` function. Which makes me wonder, is it possible to combine all sprites we need into a single mesh renderer and have the shader process a single sprite instead of multiple?

I have tried adding a random value to our function that updates the anchor points, so perhaps there's a value we could expose from the shader to use as an offset inside the update anchor function.

Doddler has a variable called `_Offset` on his renderer and shader and I'm wondering if that offset is this one we're trying to workaround here

**Comment by Jlabarca** [comment]
Removed the billboard part form the shader, it is handled by the monobehaviour script Billboard.cs again until we have a way to join all sprites quads in one

**Comment by Danil0v3s** [comment]
Thanks for that, it seems to be working a lot better now.

**Pull request merged**